Lib/_strptime.py

```diff
@@ -313,8 +313,8 @@ def __init__(self, locale_time=None):
                                 'Z'),
             '%': '%'}
         for d in 'dmyHIMS':
-            mapping['O' + d] = r'(?P<%s>\d\d|\d| \d)' % d
-        mapping['Ow'] = r'(?P<w>\d)'
+            mapping['O' + d] = r'(?P<%s>[0-9][0-9]|[0-9]| [0-9])' % d
+        mapping['Ow'] = r'(?P<w>[0-9])'
```

Comment:
This is wrong.

Reply:
It was like this before? Is it meant to match 2 digits?

```suggestion
            mapping['O' + d] = r'(?P<%s>[0-9][0-9]|[0-9]| [0-9])' % d
        mapping['Ow'] = r'(?P<w>[0-9]{1,2})'
```

Reply:
@StanFromIreland No, I believe the issue is that the `E` and `O` prefixes are locale-specific, and should match non-ASCII digits.

It is not clear to me if we would want it to match *only* the locale-specific version of this when parsing, or match anything that might be there in any locale, but in any case requiring ASCII digits here is wrong.

Reply:
Fixed. Please do document them somehow